src/bridges/erc4626/VaultBridge.sol

```diff
@@ -0,0 +1,152 @@
+        ) revert AssetMustBeEmpty();
+
+        if (isValidPair(outputAssetA.erc20Address, inputAssetA.erc20Address)) {
+            outputValueA = _enter(outputAssetA.erc20Address, totalInputValue);
+        } else if (isValidPair(inputAssetA.erc20Address, outputAssetA.erc20Address)) {
+            outputValueA = _exit(inputAssetA.erc20Address, totalInputValue);
+        } else {
+            revert ErrorLib.InvalidInput();
+        }
+    }
+
+    /**
+     * @notice Public Function used to preapprove vault pairs
+     * @param vault address of erc4626 vault
+     * @param token address of the vault asset
+    
+     */
+    function approvePair(address vault, address token) public {
+        IERC20(token).safeApprove(address(vault), 0);
+        IERC20(token).safeApprove(address(vault), type(uint256).max);
+        IERC20(token).safeApprove(address(ROLLUP_PROCESSOR), 0);
+        IERC20(token).safeApprove(address(ROLLUP_PROCESSOR), type(uint256).max);
+
+        IERC20(vault).approve(ROLLUP_PROCESSOR, type(uint256).max);
+    }
+
+    /**
+     * @notice Function which checks whether a token is a vaild asset for a given vault
+     * @dev This method is used to check whether input and output assets are matching vault share/asset pairs
+     *
+     * @param vault address of erc4626 vault
+     * @param asset address of the vault asset token
+     *
+     */
+    function isValidPair(address vault, address asset) internal returns (bool) {
+        try IERC4626(vault).asset() returns (IERC20 token) {
+            return address(token) == asset;
+        } catch {
+            return false;
+        }
+    }
+
+    /**
+     * @notice Internal Function used to stake
+     * @dev This method deposits an exact amount of asset into an erc4626 vault
+     *
+     * @param vault address of erc4626 vault     
+     * @param amount amount of an asset to be burned
+   
+     */
+    // enter by deposit exact assets
+    function _enter(address vault, uint256 amount) internal returns (uint256) {
+        return IERC4626(vault).deposit(amount, address(this));
+    }
+
+    /**
+     * @notice Internal Function used to unstake
+     * @dev This method redeems an exact number of shares into an erc4626 vault
+     *
+     * @param vault address of erc4626 vault
+     * @param amount amount of shares to be redeemed
+     *
+     */
+    function _exit(address vault, uint256 amount) internal returns (uint256) {
+        return IERC4626(vault).redeem(amount, address(this), address(this));
+    }
+
+    // @notice This function always reverts because this contract does not implement async flow.
+    function finalise(
+        AztecTypes.AztecAsset calldata inputAssetA,
+        AztecTypes.AztecAsset calldata inputAssetB,
+        AztecTypes.AztecAsset calldata outputAssetA,
+        AztecTypes.AztecAsset calldata outputAssetB,
+        uint256 interactionNonce,
+        uint64 auxData
+    )
```

Comment:
Since you are inheriting from BridgeBase please remove this function. https://github.com/AztecProtocol/aztec-connect-bridges/blob/5c32f02782d5649287c626fdd01afb7591068578/src/bridges/base/BridgeBase.sol#L54